cardano-cli/src/Cardano/CLI/Shelley/Run/Governance.hs

```diff
@@ -177,3 +221,152 @@ runGovernanceUpdateProposal (OutputFile upFile) eNo genVerKeyFiles upPprams mCos
 
   firstExceptT ShelleyGovernanceCmdTextEnvWriteError . newExceptT $ writeFileTextEnvelope upFile Nothing upProp
 
+runGovernanceCreatePoll
+  :: Text
+  -> [Text]
+  -> Maybe Word
+  -> OutputFile
+  -> ExceptT ShelleyGovernanceCmdError IO ()
+runGovernanceCreatePoll govPollQuestion govPollAnswers govPollNonce out = do
+  let poll = GovernancePoll{ govPollQuestion, govPollAnswers, govPollNonce }
+
+  let description = fromString $ "An on-chain poll for SPOs: " <> Text.unpack govPollQuestion
+  firstExceptT ShelleyGovernanceCmdTextEnvWriteError . newExceptT $
+    writeFileTextEnvelope (unOutputFile out) (Just description) poll
+
+  let metadata = asTxMetadata poll
+        & metadataToJson TxMetadataJsonDetailedSchema
+
+  let outPath = unOutputFile out
+        & Text.encodeUtf8 . Text.pack
+
+  liftIO $ do
+    BSC.hPutStrLn stderr
+      "Poll created successfully.\n\
+      \Please submit a transaction using the resulting metadata.\n"
+    BSC.hPutStrLn stdout (prettyPrintJSON metadata)
+    BSC.hPutStrLn stderr $
+      "\n\
+      \Hint (1): Use '--json-metadata-detailed-schema' and '--metadata-json-file' \
+      \from the build or build-raw commands.\n\
+      \Hint (2): You can redirect the standard output of this command to a JSON \
+      \file to capture metadata.\n\n\
+      \Note: A serialized version of the poll suitable for sharing with \
+      \participants has been generated at '" <> outPath <> "'."
+
+runGovernanceAnswerPoll
+  :: FilePath
+  -> SigningKeyFile  -- VRF or Ed25519 cold key
+  -> Maybe Word -- Answer index
```

Comment:
Replace `--` with `-- ^` to turn this into haddock comments.

Reply:
https://github.com/input-output-hk/cardano-node/compare/c018de7877eb6c513ddc029224ba2a189c4f6b2c..7adb11180e227089a5aa44845a06737efacf3401